sql: add tests for casting NULL to any scalar

otan opened the pull request:
This PR fixes a bug where geospatial types cannot be cast from NULL.
Furthermore, prevent this error by making sure any type can be cast from
NULL.

Resolves #47398.
Resolves #47395.

Release note: None

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.png" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/47434)
<!-- Reviewable:end -->

rohany reviewed (approved):
look at you contributing to the boilerplate you complained about :D

otan committed:
sql: add tests for casting NULL to any scalar

This PR fixes a bug where geospatial types cannot be cast from NULL.
Furthermore, prevent this error by making sure any type can be cast from
NULL.

Resolves #47398.
Resolves #47395.

Release note: None

otan commented:
> look at you contributing to the boilerplate you complained about :D

just playin' the game

otan commented:
bors r=rohany
